src/components/SignupModal.jsx

```diff
@@ -6,31 +6,246 @@ import {
   ModalFooter,
   ModalBody,
   ModalCloseButton,
-  Text,
+  ButtonGroup,
   Button,
   useDisclosure,
-} from "@chakra-ui/react"
+  Input,
+  FormControl,
+  FormLabel,
+} from "@chakra-ui/react";
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { SiGoogle } from "react-icons/si";
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+ } from "firebase/auth";
+import {
+  auth,
+  signInWithGoogle,
+} from '../firebaseInit';
+
 
 export function SignupModal() {
-  const { isOpen, onOpen, onClose } = useDisclosure()
-  return (
-    <>
-      <Button onClick={onOpen}>Sign in / Sign Up</Button>
+  const { isOpen: isLoginOpen, onOpen: onLoginOpen, onClose: onLoginClose } = useDisclosure()
+  const { isOpen: isSignUpOpen, onOpen: onSignUpOpen, onClose: onSignUpClose } = useDisclosure()
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [conf_password, setConfPassword] = useState('');
+  const [error, setError] = useState('');
+  const navigate = useNavigate();
+  const [user, setUser] = useState("");
+
+  useEffect( () => {
```

Comment:
are we sure that useeffect is needed here? what is the dependency that needs the re-render?

https://react.dev/reference/react/useEffect

Reply:
That is how I check and see if there is a user that is logged in. Without this the Signup/login button does not update to the correct state.